bindings/rust/wasmedge-sys/src/import_obj.rs

```diff
@@ -1,67 +1,129 @@
+        let raw_func_name: wasmedge::WasmEdge_String = StringRef::from(name.as_ref()).into();
         unsafe {
             wasmedge::WasmEdge_ImportObjectAddFunction(self.ctx, raw_func_name, (*func).ctx);
         }
-        func.registed = true;
+        func.registered = true;
+        func.ctx = std::ptr::null_mut();
     }
 
-    pub fn create_wasi(
-        args: Vec<impl AsRef<str>>,
-        envs: Vec<impl AsRef<str>>,
-        dirs: Vec<impl AsRef<str>>,
-        preopens: Vec<impl AsRef<str>>,
-    ) {
-        let cstr_args: Vec<_> = args
-            .iter()
-            .map(|arg| CString::new(arg.as_ref()).unwrap())
-            .collect();
-
-        let cstr_envs: Vec<_> = envs
-            .iter()
-            .map(|env| CString::new(env.as_ref()).unwrap())
-            .collect();
-
-        let cstr_dirs: Vec<_> = dirs
-            .iter()
-            .map(|dir| CString::new(dir.as_ref()).unwrap())
-            .collect();
-
-        let cstr_preopens: Vec<_> = preopens
-            .iter()
-            .map(|preopen| CString::new(preopen.as_ref()).unwrap())
-            .collect();
+    pub fn add_table(&mut self, name: impl AsRef<str>, table: &mut Table) {
+        unsafe {
+            wasmedge::WasmEdge_ImportObjectAddTable(
+                self.ctx,
+                wasmedge::WasmEdge_String::from(StringRef::from(name.as_ref())),
+                table.ctx,
+            );
+        }
+        table.registered = true;
+        table.ctx = std::ptr::null_mut();
+    }
 
+    pub fn add_memory(&mut self, name: impl AsRef<str>, memory: &mut Memory) {
         unsafe {
-            wasmedge::WasmEdge_ImportObjectCreateWASI(
-                cstr_args.as_ptr() as *const *const c_char,
-                args.len() as u32,
-                cstr_envs.as_ptr() as *const *const c_char,
-                envs.len() as u32,
-                cstr_dirs.as_ptr() as *const *const c_char,
-                dirs.len() as u32,
-                cstr_preopens.as_ptr() as *const *const c_char,
-                preopens.len() as u32,
+            wasmedge::WasmEdge_ImportObjectAddMemory(
+                self.ctx,
+                wasmedge::WasmEdge_String::from(StringRef::from(name.as_ref())),
+                memory.ctx,
             );
         }
+        memory.registered = true;
+        memory.ctx = std::ptr::null_mut();
+    }
+
+    pub fn add_global(&mut self, name: impl AsRef<str>, global: &mut Global) {
+        unsafe {
+            wasmedge::WasmEdge_ImportObjectAddGlobal(
+                self.ctx,
+                wasmedge::WasmEdge_String::from(StringRef::from(name.as_ref())),
+                global.ctx,
+            );
+        }
+        global.registered = true;
+        global.ctx = std::ptr::null_mut();
+    }
+
+    pub fn init_wasi<T: Iterator<Item = E>, E: AsRef<str>>(
+        &self,
+        args: Option<T>,
+        envs: Option<T>,
+        dirs: Option<T>,
```

Comment:
Wasn't this `dir` already removed from wasmedge-sdk?

Reply:
Thanks for your comments. This PR is replaced by #854, please add your comments there. Thanks again!